src/storage/cache_proxy/cluster_informer.rs

```diff
@@ -0,0 +1,533 @@
+
+        // Try to register current node to etcd
+        self.kv_engine
+            .set(
+                &KeyType::CacheNode(node_for_read.ip().to_owned()),
+                &ValueType::Json(serde_json::to_value(node_for_read.dump())?),
+                Some(SetOption {
+                    // Set lease
+                    lease: Some(lease),
+                    prev_kv: false,
+                }),
+            )
+            .await
+            .with_context(|| format!("Failed to register node to etcd"))?;
+
+        info!("register: {} to etcd success", node_for_read.ip());
+        drop(node_for_read);
+
+        // Set online
+        node.write().set_online();
+
+        // Try keep alive current node to clsuter
+        let kv_engine_clone = self.kv_engine.clone();
+        let node_clone = node.clone();
+        // let self_clone = self.clone();
+        tokio::task::spawn(async move {
+            loop {
+                info!("Keep alive node: {}", node_clone.read().ip());
+                // Sleep 30s
+                tokio::time::sleep(Duration::from_secs(10)).await;
+
+                // Keep alive
+                if let Err(e) = kv_engine_clone.lease_keep_alive(lease).await {
+                    warn!("Failed to keep alive node: {}", e);
+
+                    node_clone.write().set_offline();
+
+                    /*
+                    // Retry to keep alive node
+                    let retry_times = self_clone.retry_times.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+                    if retry_times > 3 {
+                        error!("Failed to keep alive node, retry times exceed 3");
+                        break;
+                    }
+                    // Retry to register node
+                    self_clone.register(node_clone.clone()).await.unwrap();
+                     */
+                    break;
+                }
+            }
+        });
+
+        Ok(())
+    }
+
+    /// Register current node to master and keep alive
+    pub async fn register_master(&self, node: Arc<RwLock<Node>>) -> DatenLordResult<()> {
+        // Try to get the master lock
+        let lock_key = &LockKeyType::CacheNodeMaster("/".to_string());
+        let lock = self
+            .kv_engine
+            .lock(lock_key, Duration::from_secs(MASTER_LOCK_TIMEOUT_SEC))
+            .await;
+
+        match lock {
+            Ok(_) => {
+                // Get current node info
+                let dump_node_info = node.read().dump();
+                info!("register_master: {} node try to register master", dump_node_info.clone().ip());
+
+                // Try to get master lease for current node
+                let lease = self
+                    .kv_engine
+                    .lease_grant(15)
+                    .await
+                    .with_context(|| "Failed to get lease for current node")?;
+
+                let value_key = &KeyType::CacheNodeMaster("/".to_string());
```

Comment:
Should use a prefix instead of "/",  since watch has better behavior under a prefix